Major updates of Command Line Interface

andylvua opened the pull request:
## CLI update
* Now screens are using convenient options that can be defined using implicit lambda expressions or macro definitions
* CLI now properly using shared pointers for navigation
* CLI has been moved to use the `ncurses` library, which increases portability and facilitates further development

### Additional updates:
* Raspberry-specific compilation targets to improve cross-platform compatibility. No need for restructuring the code every time 
* Debug flags. Possibility to exclude the initialization of the module during the period of active development in order to speed up the loading of the program

codefloww reviewed (commented):
forgot to include functional.h in option.h

codefloww reviewed (commented):
missing some rerendering in leaf screens

codefloww reviewed (commented):
io not working for reading number or message in leaf screens

andylvua commented:
> forgot to include functional.h in option.h

Fixed in 07703452fc31c75c92dd08cafebeba8e27ec8d34



> missing some rerendering in leaf screens

> io not working for reading number or message in leaf screens

Fixed in 42db2c9a7ebee3f13c7b417170d5b9e697584b49, f0db8a0677d530ab68e5f49429ea6e5c09d325af

codefloww commented on CMakeLists.txt:
```diff
@@ -37,4 +50,4 @@ endif()
 set(CMAKE_PREFIX_PATH "/usr/local/bin")
-target_link_libraries(phone wiringPi)
\ No newline at end of file
+#target_link_libraries(phone wiringPi)
```
maybe put wiringPi in raspberry build

andylvua replied:
Fixed in 8ebd073636dd55bdc113567c80c63849079ff53c

codefloww commented on Src/cli/cli.cpp:
```diff
@@ -43,449 +47,356 @@ bool checkNumber(std::string &number) {
-void CLI::changeScreen(std::shared_ptr<Screen> screen) {
-    currentScreen = std::move(screen);
+void CLI::changeScreen(const QString &screenName) {
+    qDebug() << "Changing screen to " << screenName;
```
remove qdebug pleaseeee

andylvua replied:
Fixed in 3812eeba4db3f395c3738a1ce1a7c82a5aa25d54

codefloww reviewed (commented):
nice
